ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/keys/KeyMonotonicityValidation.scala

```diff
@@ -41,7 +41,7 @@ private[keys] object KeyMonotonicityValidation {
         recordTime,
         transactionCommitter.buildRejectionLogEntry(
           transactionEntry,
-          RejectionReason.Inconsistent("Causal monotonicity violated"),
+          RejectionReason.InvalidLedgerTime("Causal monotonicity violated"),
```

Comment:
@rautenrieth-da do you think that's fine?

Reply:
Is that tested somewhere? I would have expected to see a corresponding test change.

Reply:
Good point, I will add a test if Robert is fine with the change

Reply:
`InvalidLedgerTime` is the correct rejection reason here.

I expect that this will change the error visible in the ledger API from `INVALID_ARGUMENT` to `ABORTED` (see [here](https://github.com/digital-asset/daml/blob/main/ledger/participant-integration-api/src/main/scala/platform/store/CompletionFromTransaction.scala#L42)), which is also the better error message (given the [respective descriptions](https://grpc.github.io/grpc/core/md_doc_statuscodes.html)).